lib/valueflow.cpp

```diff
@@ -896,7 +896,7 @@ struct ValueFlowAnalyzer : Analyzer {
 
     using ProgramState = ProgramMemory::Map;
 
-    virtual ProgramState getProgramState() const = 0;
+    virtual std::shared_ptr<ProgramState> getProgramState() const = 0;
```

Comment:
Why is this changed to a `shared_ptr`? Its never shared.

Reply:
Because it is used to construct a `ProgramMemory` object in `ValueFlowAnalyzer::evaluateInt()`.

Reply:
But you can just move it instead.

Reply:
Reverted.